tensorboard/functionaltests/basic_test.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2016 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Basic TensorBoard functional tests using WebDriver."""
+
+import os
+import subprocess
+import time
+import unittest
+from testing.web import webtest
+from selenium.webdriver.support import expected_conditions
+from selenium.webdriver.support import wait
+from selenium.webdriver.common import by
+
+
+class BasicTest(unittest.TestCase):
+
+  @classmethod
+  def setUpClass(self):
+    src_dir = os.environ["TEST_SRCDIR"]
+    print(src_dir)
+    binary = os.path.join(src_dir,
+                          "org_tensorflow_tensorboard/tensorboard/tensorboard")
+    log_dir = os.path.join(
+        src_dir,
+        "org_tensorflow_tensorboard/tensorboard/functionaltests/testdata")
+    self.process = subprocess.Popen(
+        [binary, "--port", "8000", "--logdir", log_dir])
+
+  @classmethod
+  def tearDownClass(self):
+    self.process.kill()
+    self.process.wait()
+
+  def setUp(self):
+    self.driver = webtest.new_webdriver_session()
+    self.driver.get("http://localhost:8000")
+    self.wait = wait.WebDriverWait(self.driver, 2)
+
+  def tearDown(self):
+    try:
+      self.driver.quit()
+    finally:
+      self.driver = None
+      self.wait = None
+
+  def testToolbarTitleDisplays(self):
+    self.wait.until(
+        expected_conditions.text_to_be_present_in_element((
+            by.By.CLASS_NAME, "toolbar-title"), "TensorBoard"))
+
+  def testChartAppearWhenHeadingClicks(self):
+    self.wait.until(
+        expected_conditions.element_to_be_clickable((
+            by.By.CSS_SELECTOR,
+            "tf-collapsable-pane.tf-collapsable-pane-0 button"))).click()
+    self.wait.until(
```

Comment:
If I'm reading this right, the logic for this test is, "wait until the success condition is met, then (implicitly) return." So…if the test fails, does this just hang?